### ISSUE TITLE
Fine-tuning with 10% labeled data with SwAV-learned embeddings

### ISSUE THREAD

**Issue by sayakpaul** [ayulockin/SwAV-TF#6]
@ayulockin 

From the fine-tuning notebooks ([10 epochs](https://github.com/ayulockin/SwAV-TF/blob/master/Fine_Tuning_on_Flower_Dataset.ipynb) and [40 epochs](https://github.com/ayulockin/SwAV-TF/blob/master/Fine_Tuning_on_Flower_Dataset(40_epochs).ipynb)) I have the following observations:

- With SwAV embeddings from 10 epochs, the model tends to have a pretty large overfit margin (note that this after the final fine-tuning is done):
   
   ![image](https://user-images.githubusercontent.com/22957388/90584465-4a1c7100-e1f0-11ea-9b9c-08283866f8b3.png)

Final progress (with EarlyStopping) - 

> loss: 0.5366 - acc: 0.8120 - val_loss: 1.8524 - val_acc: 0.5000

- With the same embeddings along with augmentation, the model seems to recover the large gap - 
   
   ![image](https://user-images.githubusercontent.com/22957388/90584635-9ebfec00-e1f0-11ea-848b-f8161b3273ed.png)

Final progress (with EarlyStopping) - 

> loss: 0.9433 - acc: 0.6104 - val_loss: 1.2685 - val_acc: 0.5455

- Going in the same order of experiments, with embeddings from 40 epochs, the following is what we get after the final fine-tuning (without any augmentation) - 

  ![image](https://user-images.githubusercontent.com/22957388/90584786-05450a00-e1f1-11ea-90d9-8feb1f8b54c5.png)

Final progress - 

> loss: 1.3076 - acc: 0.5613 - val_loss: 1.7242 - val_acc: 0.4800

**Note** that the model does not suffer from large overfitting gap in this case. 

- With augmentation, we get - 

   ![image](https://user-images.githubusercontent.com/22957388/90584891-38879900-e1f1-11ea-8053-502b15f5793e.png)

Final progress - 

> loss: 0.9239 - acc: 0.6621 - val_loss: 1.3685 - val_acc: 0.5291

This performance is almost similar to what we got in this setting with the embeddings from 10 epochs.
